reporter/src/main/kotlin/model/EvaluatedModel.kt

```diff
@@ -19,7 +19,284 @@
+        }
+
+        private val JSON_MAPPER = JsonMapper().apply { MAPPER_CONFIG() }
+
+        fun create(input: ReporterInput): EvaluatedModel = EvaluatedModelBuilder(input).build()
+    }
+
+    fun toJson(): String {
+        val tree = JSON_MAPPER.valueToTree<ObjectNode>(this)
+        tree.forEach { node ->
+            if (node is ArrayNode) {
+                if (!node.isEmpty && node[0].has("_id")) {
+                    val sortedChildren =
+                        node.elements().asSequence().toList().sortedBy { it["_id"].intValue() }.toList()
+                    node.removeAll()
+                    node.addAll(sortedChildren)
+                }
+            }
+        }
+        return tree.toPrettyString()
+    }
+}
+
+@JsonIgnoreProperties(value = ["paths"], allowGetters = true)
+data class EvaluatedPackage(
+    val id: Identifier,
+    val isProject: Boolean,
+    val definitionFilePath: String,
+    val purl: String = id.toPurl(),
+    val declaredLicenses: List<LicenseId>,
+    val declaredLicensesProcessed: EvaluatedProcessedDeclaredLicense,
+    val detectedLicenses: Set<LicenseId>,
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val concludedLicense: SpdxExpression? = null,
+    val description: String,
+    val homepageUrl: String,
+    val binaryArtifact: RemoteArtifact,
+    val sourceArtifact: RemoteArtifact,
+    val vcs: VcsInfo,
+    val vcsProcessed: VcsInfo = vcs.normalize(),
+    val curations: List<PackageCurationResult>,
+    @JsonIdentityReference(alwaysAsId = true)
+    var paths: MutableList<EvaluatedPackagePath> = mutableListOf(),
+    val levels: SortedSet<Int>,
+    val scopes: MutableSet<ScopeName>,
+    val scanResults: List<EvaluatedScanResult>,
+    val findings: List<EvaluatedFinding>,
+    val isExcluded: Boolean,
+    val pathExcludes: List<PathExclude>,
+    val scopeExcludes: List<ScopeExclude>,
+    val issues: List<EvaluatedOrtIssue>
+)
+
+data class EvaluatedPackagePath(
+    val pkg: EvaluatedPackage,
+    val project: EvaluatedPackage,
+    val scope: ScopeName,
+    val path: List<EvaluatedPackage>
+)
+
+data class EvaluatedScanResult(
+    val provenance: Provenance,
+    val scanner: ScannerDetails,
+    val startTime: Instant,
+    val endTime: Instant,
+    val fileCount: Int,
+    val packageVerificationCode: String,
+    val issues: List<EvaluatedOrtIssue>
+)
+
+enum class EvaluatedFindingType {
+    COPYRIGHT, LICENSE
+}
+
+data class EvaluatedFinding(
+    val type: EvaluatedFindingType,
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val license: LicenseId?,
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val copyright: CopyrightStatement?,
+    val path: String,
+    val startLine: Int,
+    val endLine: Int,
+    val scanResult: EvaluatedScanResult
+)
+
+data class CopyrightStatement(
+    val statement: String
+)
+
+data class LicenseId(
+    val id: String
+)
+
+data class ScopeName(
+    val name: String
+)
```

Comment:
Why are the above 3 types of the properties each `wrapped` ?

Reply:
To allow Jackson to generate IDs for them. I will add a code comment.

Reply:
Done.